### DISCUSSION TITLE
Optimizing products with specific tables in the database such as HPOS for orders, to increase speed on the site and admin

### DISCUSSION THREAD

**Discussion by paulvek** [woocommerce/woocommerce#44350] [General]
There is still no possibility for WooCommerce to store product information and specifications in completely separate tables from the default WooCommerce post types, as with HPOS. To increase the speed of the site and the system, and to make its access and development better and faster for the site administrator and developer. There is still much more important work to be done.

We are still waiting for the optimization of the products section in the admin and client, the optimization of the products in the database, having separate tables for descriptions and tags and product categories. Tables like this wp_woocommerce_products_desc and wp_woocommerce_products_tags and wp_woocommerce_products_category and etc. that are easily available And WooCommerce becomes lighter and faster.

**Comment by lsinger** [comment]
Thank you @paulvek! While this is not on our roadmap for now, we do evaluate our roadmap continuously and can add it if there is lot of demand and benefits. Would you be able to share a few more details around your use case and the challenges you've been seeing? 

Also pinging @wcstaley for awareness.

**Reply by paulvek** [reply]
Thank you @lsinger. I think it is clear. All the challenges are related to the database not being organized and the difficulty of accessing the specific information of the products, reducing the speed of saving and downloading (because WordPress needs to find the information needed by the products from among the pile of information it does not need), difficult expandability.
WordPress is open source and free, and so is WooCommerce. Maybe we need to personally build a system based on our own needs in our own organization by removing some features from WordPress and WooCommerce adding some new features and using it personally for our own organization, or even switching to another system. To move, in this situation, everything seems impossible, although it is possible, but very difficult and bad.
Therefore, the development and transfer are difficult and make the work very complicated and nerve-wracking. And there is also the issue of performance. Due to the mixing of tables and data, the speed decreases. Just some time ago, someone came to us and wanted to program a completely dedicated system for his site, and he was supposed to use the tables of WooCommerce products and orders. so that they can finally transfer products, orders, and customer information.
With the launch of HPOS, the problem of orders has been solved to a large extent, but the products still need to be optimized with a similar possibility. And really, if any expert programmer in the world wants to comment, this is a basic requirement. If you pay attention, it is getting bolder every day and its requests are increasing.
Just do a search here and in the WordPress repository and see comments and opinions and other places. Although many want such a thing but do not express it, or do not have hope that they want to make a change. However, I need representation. Why don't you have it in your work schedule? I think it's better to include it for several reasons. Not only now. Rather, it would have been better to have thought about it and worked on it a long time ago. Thank you very much and thank you for your efforts...

**Comment by lkraav** [comment]
https://github.com/woocommerce/woocommerce-product-tables-feature-plugin got stalled at one point, but I'm sure Issues discussions there reveal where and what exactly.

**Reply by paulvek** [reply]
bad news, Because of the difficulty of the work, they decided to leave it, so that they might be comfortable!

**Comment by vedanshujain** [comment]
Just noting that working on HPOS-like project for products comes with its own caveats such as everyone in the ecosystem would need to audit and possibly modify their code for compatibility with the new system. It also generally increases the support burden in the start. But most importantly, there are risks to merchant operations that we would like to manage.

This is to say that the bar is pretty high for us to start a project of this complexity. Not saying that we won't but just that this is under active discussion, so use cases around specific problems and potential improvements helps us figure out both - whether doing something like this makes sense and also what the solution should look like.

So if we have more specifics around the kind of problems at different scales that we intend to solve, it will definitely help.

**Comment by rpldom** [comment]
> There is still no possibility for WooCommerce to store product information and specifications in completely separate tables from the default WooCommerce post types, as with HPOS. To increase the speed of the site and the system, and to make its access and development better and faster for the site administrator and developer. There is still much more important work to be done.
> 
> We are still waiting for the optimization of the products section in the admin and client, the optimization of the products in the database, having separate tables for descriptions and tags and product categories. Tables like this wp_woocommerce_products_desc and wp_woocommerce_products_tags and wp_woocommerce_products_category and etc. that are easily available And WooCommerce becomes lighter and faster.

I definitely second this!

**Comment by FunkyCss** [comment]
I read so many comments on forums and groups about this, but you say there is no demand? Insane
